### PR TITLE
PPTP-893 Use 'InternalId' to save registration progress 

### PR DESCRIPTION
**Pull request by PauloPortugal**
Also Increase code coverage threshold.

### CODE REVIEW

**Review by tmc-ee — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala [hunk 1]
```diff
@@ -65,18 +61,16 @@ class Authenticator @Inject() (override val authConnector: AuthConnector, cc: Co
...
-      hasEnrolment(enrolments, pptEnrolmentIdentifierName) match {
-        case Some(utr) => Future.successful(Right(AuthorizedRequest(utr.value, request)))
-        case _ =>
-          val msg = "Unauthorised access. User without PPT Id."
-          logger.error(msg)
-          Future.successful(Left(ErrorResponse(UNAUTHORIZED, msg)))
-      }
+    authorised().retrieve(internalId) {
+      case None =>
+        val msg = "Unauthorised access. User without PPT Id."
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala:70]: Is this message a little misleading? Are we using a PPT id or just a generic user id?

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala [hunk 2]
```diff
@@ -65,18 +61,16 @@ class Authenticator @Inject() (override val authConnector: AuthConnector, cc: Co
...
+        logger.error(msg)
+        Future.successful(Left(ErrorResponse(UNAUTHORIZED, msg)))
+      case Some(id) => Future.successful(Right(AuthorizedRequest(id, request)))
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala:73]: Is this correct? We are permitting access to any user to the PPT registration? It feels a little different to what we had previously where a user needed to have a "ppt enrolment key".

**Review by RyanKellyHMRC — COMMENTED** [review]
I think what's been done if probably alright but I'm just asking some questions to double check before approving.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala (hunk 1 above)
Comment (PauloPortugal) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala:70]: It is misleading. It should read user without HMRC Internal Id (which means he is not authorised)
Let me correct the wording.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala (hunk 2 above)
Comment (PauloPortugal) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/actions/Authenticator.scala:73]: > It feels a little different to what we had previously where a user needed to have a "ppt enrolment key".
For registrations we won't have to check any enrollment at this stage. 

> We are permitting access to any user to the PPT registration?
Yes any limited company or sole trader would have access to the PPT registration service, regardless of their (other) HMRC enrolments.

**Review by RyanKellyHMRC — APPROVED** [review]

**Pull request merged**